internal/mage/util/util.go

```diff
@@ -79,3 +88,118 @@ func DaggerBinary(c *dagger.Client) *dagger.File {
 		}).
 		File("./bin/cloak")
 }
+
+const (
+	sdkHelper      = "dagger-sdk-helper"
+	buildkitRepo   = "github.com/moby/buildkit"
+	buildkitBranch = "v0.10.5"
+)
+
+func DevEngineContainer(c *dagger.Client, arches, oses []string) []*dagger.Container {
+	buildkitRepo := c.Git(buildkitRepo).Branch(buildkitBranch).Tree()
+
+	platformVariants := make([]*dagger.Container, 0, len(arches))
+	for _, arch := range arches {
+		buildkitBase := c.Container(dagger.ContainerOpts{
+			Platform: dagger.Platform("linux/" + arch),
+		}).Build(buildkitRepo)
+		for _, os := range oses {
+			// include each helper for each arch too in case there is a
+			// client/server mismatch
+			for _, arch := range arches {
+				helperBin := GoBase(c).
+					WithEnvVariable("GOOS", os).
+					WithEnvVariable("GOARCH", arch).
+					Exec(dagger.ContainerExecOpts{
+						Args: []string{"go", "build", "-o", "./bin/" + sdkHelper, "-ldflags", "-s -w", "/app/cmd/sdk-helper"},
+					}).
+					File("./bin/" + sdkHelper)
+				buildkitBase = buildkitBase.WithFS(
+					buildkitBase.FS().WithFile("/usr/bin/"+sdkHelper+"-"+os+"-"+arch, helperBin),
+				)
+			}
+		}
+		platformVariants = append(platformVariants, buildkitBase)
+	}
+
+	return platformVariants
+}
+
+func DevEngine(ctx context.Context, c *dagger.Client) (string, error) {
```

Comment:
maybe we should protect this behind a `sync.Once`?

Reply:
Yeah good call, will update